tasks/verify-enterprise-contract.yaml

```diff
@@ -0,0 +1,123 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: verify-enterprise-contract
+  annotations:
+    tekton.dev/displayName: Verify Enterprise Contract
+    tekton.dev/pipelines.minVersion: "0.19"
+    tekton.dev/tags: cosign, chains, signature, opa
+  labels:
+    app.kubernetes.io/version: "0.1"
+
+spec:
+  description: Verify the enterprise contract is met
+  params:
+    - name: IMAGE_REF
+      type: string
+      description: Image reference to verify
+
+    - name: PUBLIC_KEY
+      type: string
+      description: >-
+        Public key used to verify signatures. Must be a valid k8s cosign
+        reference, e.g. k8s://my-space/my-secret where my-secret contains
+        the expected cosign.pub attribute.
```

Comment:
Not allowing an inline public key can lead to issues when the signing key is rolled over as we assume that the signature was created with the same key pair that is stored in both secrets: `signing-secrets` for chains controller and the `...-cosign-public-key` secret containing only the public key. There is a window where the signature is created using previous key pair and both secrets have been updated to the newest key. Running and retrying the task can become problematic as the time between signing the image/attestation and verifying the signature increases.